src/state/opamEnv.ml

```diff
@@ -639,14 +639,21 @@ let update_user_setup root ?dot_profile shell =
     OpamStd.Option.iter (fun f -> update_dot_profile root f shell) dot_profile
   )
 
+let status st =
+  OpamStateConfig.update ~no_env_notice:false ();
```

Comment:
Don't do side effect in a function that should be pure according to what the interface lets assume.
Either add an optional argument to `is_up_to_date`, or warn against `no_env_notice` in the .mli.